### DISCUSSION TITLE
add option to specify the ping payload

### DISCUSSION THREAD

**Discussion by pmeier** [frankie567/httpx-ws#86] [Ideas]
I'm wrapping a WS endpoint that I have no control over. Unfortunately, it requires a specific payload when pinging and locks up when sending a random one. Using the high level API, i.e. `connect_ws` there is no way to specify a "payload factory" or the like. The only regular option I currently have to achieve my goal is to ditch the high level API, instantiate the `WebSocketSession` manually and override the `.ping()` method. However, in the process I lose the setup code done by `_connect_ws`:

https://github.com/frankie567/httpx-ws/blob/3a450c0763fad5664d40187e8e2cc41d8c663e4a/httpx_ws/_api.py#L1204-L1209

Thus, I'd have to vendor this as well. Thus, I currently fall back to monkey-patching `.ping`.

Would it be possible to add a "payload factory" to the existing ping parameters?

https://github.com/frankie567/httpx-ws/blob/3a450c0763fad5664d40187e8e2cc41d8c663e4a/httpx_ws/_api.py#L1195-L1200

**Comment by frankie567** [comment]
I see, makes sense 🤔 

Instead of exposing a specific argument just for `ping`, what would you think about the possibility to pass directly websocket session class as argument of `aconnect_ws`? In your case, it would look like this:

```py
class CustomPingAsyncWebSocketSession(AsyncWebSocketSession):
    async def ping(self, payload: bytes = b"custom_ping_payload") -> anyio.Event:
        return await super().ping(payload)

async with aconnect_ws("http://localhost:8000/ws", session_class=CustomPingAsyncWebSocketSession) as ws:
    message = await ws.receive_text()
```

This approach enables the possibility to override basically anything without losing the niceties of `aconnect_ws`.

**Reply by pmeier** [reply]
Sure, works for me as well

**Reply by frankie567** [reply]
Opened #87 to track this
